packages/nx/src/project-graph/utils/project-configuration-utils.ts

```diff
@@ -518,31 +513,25 @@ export function readProjectConfigurationsFromRootMap(
   const projectRootsWithNoName: string[] = [];
 
   for (const root in projectRootMap) {
-    const configuration = projectRootMap[root];
+    const project = projectRootMap[root];
     // We're setting `// targets` as a comment `targets` is empty due to Project Crystal.
     // Strip it before returning configuration for usage.
-    if (configuration['// targets']) delete configuration['// targets'];
-    if (!configuration.name) {
-      try {
-        const { name } = readJsonFile(join(root, 'package.json'));
-        if (!name) {
-          throw new Error("No name found for project at '" + root + "'.");
+    if (project['// targets']) delete project['// targets'];
+
+    try {
+      validateProject(project, projects);
+    } catch (e) {
+      if (isProjectsWithNoNameError(e)) {
+        projectRootsWithNoName.push(...e.projectRoots);
+      } else if (isProjectsWithConflictingNamesError(e)) {
+        const newConflicts = e.conflicts;
+        for (const [name, roots] of newConflicts) {
+          const conflictsArray = conflicts.get(name) ?? [];
+          conflictsArray.push(...roots);
```

Comment:
Add some tests for these errors. I'm curious how 3+ projects with the same name are handled.